edge-modules/edge-dashboard/src/state.rs

```diff
@@ -0,0 +1,108 @@
+use std::fs;
+use std::io::Result;
+use std::env;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum State {
+    Manual,
+    NotProvisioned,
+    NotInstalled,
+    // DPS
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Device {
+    state: State,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    hub_name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    device_id: Option<String>,
+}
+
+impl Device {
+    // makes a new Device struct based on the state
+    pub fn new(state: &str, con_string: String) -> Device {
+        match state {
+            "manual"          => Device::manual_rep(con_string),
+            "not provisioned" => Device::default(State::NotProvisioned),
+            // "dps"          => ,
+            _                 => Device::default(State::NotInstalled),
+        }
+    }
+
+    // returns the manual state representation of device
+    pub fn manual_rep(con_string: String) -> Device {
```

Comment:
For parsing connection strings consider using something like the following:
```rust
pub fn parse_str(query: &str, pair_delim: char, kv_delim: char) -> HashMap<&str, &str> {
    query
        .split(pair_delim)
        .filter_map(|seg| {
            if seg.is_empty() {
                None
            } else {
                let mut tokens = seg.splitn(2, kv_delim);
                if let Some(key) = tokens.next() {
                    let val = tokens.next().unwrap_or("");
                    Some((key, val))
                } else {
                    // if there's no key then we ignore this token
                    None
                }
            }
        })
        .collect()
}
```
With this function now you can do the following:
```rust
let map = parse_str(inp, ';', '=');
let hub_name = map["HostName"];
let device_id = map["DeviceId"];
let access_key = map["SharedAccessKey"];
```

Reply:
There's a variant of this in `edgelet/edgelet-utils/src/lib.rs` called `parse_query`. Maybe we can adapt that to look like the above.